### PR TITLE
zfs: 2.1.5 → 2.1.6

### LINKED ISSUE #194311: zfs-2.1.6
###### Checklist

https://github.com/openzfs/zfs/releases/tag/zfs-2.1.6

- [x] Checked the [nixpkgs master branch](https://github.com/NixOS/nixpkgs)

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

###### Project name
`nix search` name:

current version: 2.1.5
desired version: 2.1.6

###### Notify maintainers

maintainers: @hmenke @jcumming @wizeman @globin @jonringer 

###### Note for maintainers

Please tag this issue in your merge request.

The kernel version needs to be bumped to 5.19.x. It's not ready for 6.0.x.

### PR DESCRIPTION
**Pull request by toastal**
###### Description of changes

fixes #194311

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/zfs/default.nix [hunk 1]
```diff
@@ -216,13 +216,14 @@ in {
...
     # check the release notes for compatible kernels
-    kernelCompatible = kernel.kernelOlder "5.19";
-    latestCompatibleLinuxPackages = linuxPackages_5_15;
+    kernelCompatible = kernel.kernelOlder "5.20";
+    latestCompatibleLinuxPackages = linuxPackages_5_19;
 
     # this package should point to the latest release.
-    version = "2.1.5";
+    version = "2.1.6";
+    rev = "6a6bd493988c75331deab06e5352a9bed035a87d";
```
Comment (lovesegfault) [inline pkgs/os-specific/linux/zfs/default.nix:224]: Why do you need this? There's a 2.1.6 tag: https://github.com/openzfs/zfs/releases/tag/zfs-2.1.6

Comment (toastal) [reply pkgs/os-specific/linux/zfs/default.nix:224]: It's the same revision. I'm not sure on the protocol here, more I just want to get the ball rolling on support.

Comment (toastal) [reply pkgs/os-specific/linux/zfs/default.nix]: But it looks like I can just skip the revision in this case

**Review by hmenke — CHANGES_REQUESTED** [review]
Your PR does not evaluate because you forgot to update `zfsUnstable`.

**Comment by toastal** [comment]
Ah, closed by#194405

Cool. I'm happy someone covered it!

**Pull request closed without merging**